From d62d2ffc71194068af509ec3285ecd6823d883fb Mon Sep 17 00:00:00 2001
From: Andrea Claudi <aclaudi@redhat.com>
Date: Mon, 29 Apr 2019 20:07:22 +0200
Subject: [PATCH] ipntable: No need to check and assign to parms_rta

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1465646
Upstream Status: iproute2.git commit 2869262144271

commit 28692621442710f4a67fe33742f56efc582ee33a
Author: Phil Sutter <phil@nwl.cc>
Date:   Thu Aug 17 19:09:26 2017 +0200

    ipntable: No need to check and assign to parms_rta

    This variable is initialized at declaration and nowhere else does any
    assignment to it happen, so just drop the check.

    Signed-off-by: Phil Sutter <phil@nwl.cc>
---
 ip/ipntable.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/ip/ipntable.c b/ip/ipntable.c
index 65063321c85f8..ae8c74ead2cb8 100644
--- a/ip/ipntable.c
+++ b/ip/ipntable.c
@@ -202,8 +202,6 @@ static int ipntable_modify(int cmd, int flags, int argc, char **argv)
 			if (get_u32(&queue, *argv, 0))
 				invarg("\"queue\" value is invalid", *argv);
 
-			if (!parms_rta)
-				parms_rta = (struct rtattr *)&parms_buf;
 			rta_addattr32(parms_rta, sizeof(parms_buf),
 				      NDTPA_QUEUE_LEN, queue);
 			parms_change = 1;
-- 
2.21.0

